app/helpers/application_helper/button/host_introspect_provide.rb

```diff
@@ -0,0 +1,17 @@
+class ApplicationHelper::Button::HostIntrospectProvide < ApplicationHelper::Button::Basic
+  needs :@record
+
+  def visible?
+    proper_record? && host_manageable?
+  end
+
+  private
+
+  def proper_record?
+    @record.class == ManageIQ::Providers::Openstack::InfraManager::Host
```

Comment:
@vecerek I think, that this is something we want to avoid. Assuming if record has some class, so have some capabilities. There is support mixin for this. But I am not really sure about this. @Ladas probably know more.